generic-tabs/GenericTabs.js

```diff
@@ -35,11 +35,19 @@ export class GenericTabs extends SelectedMixin(BatchingElement) {
     return {
       selectors: {
         tabs: {
-          selector: el => el.querySelectorAll("[slot='tab']"),
+          selector: el =>
+            Array.from(el.children).filter(node =>
+              node.matches('h1, h2, h3, h4, h5, h6, [slot="tab"]'),
+            ),
           focusTarget: true,
         },
         panels: {
-          selector: el => el.querySelectorAll("[slot='panel']"),
+          selector: el =>
+            Array.from(el.children).filter(
+              node =>
+                node.matches('h1 ~ *, h2 ~ *, h3 ~ *, h4 ~ *, h5 ~ *, h6 ~ *, [slot="panel"]') &&
+                !node.matches('h1, h2, h3, h4, h5, h6, [slot="tab"]'),
```

Comment:
is this for like, if they are out of order or?

Reply:
This preserves the existing functionality for when there are no heading children and everything is explicitly slotted. This selector targets any sibling that comes after a heading, or any slotted panel. The second part then removes any siblings that are already a heading or an explicitly slotted tab. The second part could also be written like `!(node instanceof HTMLHeadingElement || node.slot === 'tab')`.

Reply:
Ah I see